### ISSUE TITLE
Handling left-truncated or left-censored data?

### ISSUE THREAD

**Issue by buttrey** [imbs-hl/ranger#328]
Hi. I have left-truncated data. Normally I would express this by calling Surv (time1, time2, event, type = "counting"), where time1 is the time at which I first saw this observation, time2 is the event or censoring time, and event describes the event. [I'm hoping that type = "counting" specifies that time1 is a truncation time, rather than a left-censoring time.] But although ranger runs with this setup, it produces prediction times that are uniformly 1. Consider the example on the last line of the help page for the pbcseq data set in the survival library. With a little modification, like explicitly adding a column named log.bili, we see this:

rf <- ranger(Surv(time1, time2, event) ~ age + sex + log.bili, data = pbcseq)
but then
all (predict (rf, data = pbcseq)$preds == 1) # produces TRUE

Every prediction is always 1. (I get the same rtesult with type="counting.") That seems off. Is the story that ranger is not (yet) set up to handle this flavor of Surv() object?

Thanks,
Sam Buttrey

**Comment by mnwright** [comment]
> Is the story that ranger is not (yet) set up to handle this flavor of Surv() object?
Exactly. For now we have to check the `Surv` object and produce an error for your example. 

Is there any literature on RF for left censoring or truncation?

**Comment by mnwright** [comment]
Thanks. I probably won't have the time to implement it soon. However, I'm happy to help if someone wants to do so.

**Comment by buttrey** [comment]
Sure, I understand. Thanks for the replies.

**Comment by osorochynskyi** [comment]
Hello,

Just letting anyone who wants to fit RF to left-truncated data know that the inability to explicitly take into account left-truncation can be offset by instead modeling on time-on-study and including the left-truncation time as a covariable. 

With this approach, the original example (`ranger(Surv(time1, time2, event) ~ age + sex + log.bili, data = pbcseq)`) would become 
`ranger(Surv(time2 - time1, event) ~ time1 + age + sex + log.bili, data = pbcseq)`.

Although this may seem irksome when coming from parametric or semi-parametric models, in practice it gives reasonable results. In fact, in medical literature, most studies adapt this approach even when working with parametric models.

All in all, the inability to model left-truncation seems to be a minor hindrance, and that's the reason why no one bothers to implement it.

PS Thanks for the great package !
